### PR TITLE
dispatcher: update to use new internal header map types

### PR DESCRIPTION
**Pull request by goaway**
Description: Update to use new internal types in Envoy.
Risk: Low
Testing: Unit and CI

Signed-off-by: Mike Schore <mike.schore@gmail.com>

### CODE REVIEW

File: library/common/http/header_utility.h [hunk 1]
```diff
@@ -17,12 +17,20 @@ namespace Utility {
 std::string convertToString(envoy_data s);
 
 /**
- * Transform envoy_headers to HeaderMap.
+ * Transform envoy_headers to RequestHeaderMap.
+ * This function copies the content.
+ * @param headers, the envoy_headers to transform.
+ * @return HeaderMapPtr, the HeaderMap 1:1 transformation of the headers param.
```
Comment (rebello95) [inline library/common/http/header_utility.h]:
```suggestion
 * @return RequestHeaderMapPtr, the RequestHeaderMap 1:1 transformation of the headers param.
```

File: library/common/http/header_utility.h [hunk 2]
```diff
@@ -17,12 +17,20 @@ namespace Utility {
...
+ */
+RequestHeaderMapPtr toRequestHeaders(envoy_headers headers);
+
+/**
+ * Transform envoy_headers to RequestHeaderMap.
  * This function copies the content.
  * @param headers, the envoy_headers to transform.
  * @return HeaderMapPtr, the HeaderMap 1:1 transformation of the headers param.
```
Comment (rebello95) [inline library/common/http/header_utility.h]:
```suggestion
 * @return RequestTrailerMapPtr, the RequestTrailerMap 1:1 transformation of the headers param.
```

File: library/common/http/header_utility.h [hunk 3]
```diff
@@ -17,12 +17,20 @@ namespace Utility {
...
  */
-HeaderMapPtr toInternalHeaders(envoy_headers headers);
+RequestTrailerMapPtr toRequestTrailers(envoy_headers headers);
```
Comment (rebello95) [inline library/common/http/header_utility.h]: nit: might also make sense to rename `headers` to `trailers` now that we have an explicit `RequestTrailerMapPtr` type

File: library/common/http/header_utility.h [hunk 4]
```diff
@@ -17,12 +17,20 @@ namespace Utility {
...
- * Transform envoy_headers to HeaderMap.
+ * Transform envoy_headers to RequestHeaderMap.
+ * This function copies the content.
+ * @param headers, the envoy_headers to transform.
+ * @return RequestHeaderMapPtr, the RequestHeaderMap 1:1 transformation of the headers param.
+ */
+RequestHeaderMapPtr toRequestHeaders(envoy_headers headers);
+
+/**
+ * Transform envoy_headers to RequestHeaderMap.
```
Comment (rebello95) [inline library/common/http/header_utility.h]:
```suggestion
 * Transform envoy_headers to RequestTrailerMap.
```

File: library/common/http/header_utility.h (hunk 3 above)
Comment (rebello95) [reply library/common/http/header_utility.h]: Thoughts?

File: test/common/http/header_utility_test.cc [hunk 5]
```diff
@@ -66,6 +75,47 @@ TEST(HeaderDataConstructorTest, FromCToCpp) {
...
+  *sentinel = 0;
+  for (size_t i = 0; i < trailers.size(); i++) {
+    header_array[i] = {
+        envoyTestString(trailers[i].first, sentinel),
+        envoyTestString(trailers[i].second, sentinel),
+    };
+  }
+
+  envoy_headers c_trailers = {static_cast<envoy_header_size_t>(trailers.size()), header_array};
+  // This copy is used for assertions given that envoy_trailers are released when toInternalHeaders
```
Comment (rebello95) [inline test/common/http/header_utility_test.cc]: `toInternalHeaders` doesn't exist anymore

**Comment by goaway** [comment]
good catches - thanks @rebello95!

**Pull request merged**